### PR TITLE
Cgmes conversion. Define structural ratio in Lines as needed.

### PR DESCRIPTION
**Pull request by marqueslanauja**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**
A `structuralRatio` is defined in flow calculation when `nominalVoltages` at both ends of a `Line` are different.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

This situation appears in boundary lines between countries.

### CODE REVIEW

File: iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTestData.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import com.powsybl.iidm.network.Line;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.iidm.network.Terminal.BusView;
+
+/**
+ *
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+public class BranchTestData {
```
Comment (MioRtia) [inline iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTestData.java]: This class, its constructor and its methods can be private-package

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java [hunk 2]
```diff
@@ -159,6 +155,18 @@ public BranchData(Line line, double epsilonX, boolean applyReactanceCorrection)
         mainComponent1 = bus1 != null ? bus1.isInMainConnectedComponent() : connectableMainComponent1;
         mainComponent2 = bus2 != null ? bus2.isInMainConnectedComponent() : connectableMainComponent2;
 
+        if (isStructuralRatioLineNeeded(line)) {
+            rho1 = 1.0 / structuralRatioEnd1(line);
+            rho2 = 1f;
```
Comment (annetill) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java]: Some duplicated code here, a small factorization could be done.

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java [hunk 3]
```diff
@@ -211,6 +219,32 @@ public BranchData(TwoWindingsTransformer twt, int phaseAngleClock, double epsilo
...
+
+        double nominalV1 = line.getTerminal1().getVoltageLevel().getNominalV();
+        double nominalV2 = line.getTerminal2().getVoltageLevel().getNominalV();
+
+        if (nominalV1 == 0.0 || Double.isNaN(nominalV1)) {
+            return false;
+        }
+        if (nominalV2 == 0.0 || Double.isNaN(nominalV2)) {
+            return false;
+        }
```
Comment (MioRtia) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java]: You can merge this into a big `if`

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java [hunk 4]
```diff
@@ -211,6 +219,32 @@ public BranchData(TwoWindingsTransformer twt, int phaseAngleClock, double epsilo
         computeValues();
     }
 
+    private boolean isStructuralRatioLineNeeded(Line line) {
+        if (line.getTerminal1() == null || line.getTerminal1().getVoltageLevel() == null) {
+            return false;
+        }
+        if (line.getTerminal2() == null || line.getTerminal2().getVoltageLevel() == null) {
+            return false;
+        }
```
Comment (MioRtia) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java]: You can merge this into a big `if`

File: iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTestData.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    private Line line;
+    private Terminal terminal1;
+    VoltageLevel voltageLevel1;
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTestData.java]: Done

File: iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTestData.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    private Terminal terminal2;
+    VoltageLevel voltageLevel2;
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTestData.java]: Done

File: iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTestData.java (hunk 1 above)
Comment (marqueslanauja) [reply iidm/iidm-api/src/test/java/com/powsybl/iidm/network/util/BranchTestData.java]: Done

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java (hunk 3 above)
Comment (marqueslanauja) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java]: Done

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java (hunk 4 above)
Comment (marqueslanauja) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java]: Done

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java (hunk 2 above)
Comment (marqueslanauja) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java]: Done

**Pull request merged**